examples/api/lib/widgets/overlay/overlay.0.dart

```diff
@@ -0,0 +1,244 @@
+                            Icon(
+                              Icons.arrow_downward,
+                              color: Colors.orange
+                            ),
+                          ],
+                        );
+                      default:
+                        return const Text('No page selected.');
+                    }
+                  }),
+                  SizedBox(
+                    width: MediaQuery.of(context).size.width / 3,
+                    height: 80.0,
+                    child: Center(
+                      child: Container(
+                        decoration: BoxDecoration(
+                          border: Border.all(
+                            color: borderColor,
+                            width: 4.0,
+                          ),
+                        ),
+                      ),
+                    ),
+                  ),
+                ],
+              ),
+            ),
+          );
+        },
+      );
+      // Add the OverlayEntry to the Overlay.
+      Overlay.of(context)!.insert(overlayEntry!);
+    }
+  }
+
+  // Remove the OverlayEntry.
+  void removeHighlightOverlay() {
+    if (overlayEntry != null) {
+      overlayEntry!.remove();
+      overlayEntry = null;
+    }
+  }
+
+  @override
+  void dispose() {
+    // Make sure to remove OverlayEntry when the widget is disposed.
+    removeHighlightOverlay();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Overlay Sample'),
+      ),
+      bottomNavigationBar: NavigationBar(
+        selectedIndex: currentPageIndex,
+        destinations: const <NavigationDestination>[
+          NavigationDestination(
+            icon: Icon(Icons.explore),
+            label: 'Explore',
+          ),
+          NavigationDestination(
+            icon: Icon(Icons.commute),
+            label: 'Commute',
+          ),
+          NavigationDestination(
+            selectedIcon: Icon(Icons.bookmark),
+            icon: Icon(Icons.bookmark_border),
+            label: 'Saved',
+          ),
+        ],
+      ),
+      body: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: <Widget>[
+          Text(
+            'Use Overlay to highlight a NavigationBar destination',
+            style: Theme.of(context).textTheme.bodyMedium,
+          ),
+          const SizedBox(height: 20.0),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              // This creates a highlight Overlay for
+              // the Explore item.
+              ElevatedButton(
+                onPressed: () {
+                  createHighlightOverlay(
+                    alignment: Alignment.bottomLeft,
+                    borderColor: Colors.red,
+                  );
+                  setState(() {
+                    currentPageIndex = 0;
```

Comment:
This works but preferably `currentPageIndex` should be updated before the overlay entry.

Reply:
Done